src/test/java/org/cactoos/experimental/TimeoutTest.java

```diff
@@ -0,0 +1,251 @@
+                1L,
+                TimeUnit.SECONDS,
+                () -> {
+                    throw new IllegalStateException("Something went wrong");
+                }
+            ).iterator().next(),
+            new Throws<>(
+                new StringContains("java.lang.IllegalStateException: Something went wrong"),
+                CompletionException.class
+            )
+        ).affirm();
+    }
+
+    /**
+     * Execute the tasks concurrently using {@link Timeout} when
+     *  {@link ExecutorService} was initiated by {@link Timeout} itself.
+     */
+    @Test
+    public void containsValuesWithInlineExecutorService() {
+        this.repeat(
+            arg -> new Assertion<>(
+                // @checkstyle LineLength (1 line)
+                "contains results from the callables when using the inline executor service",
+                new Timeout<String>(
+                    3,
+                    1L,
+                    TimeUnit.SECONDS,
+                    () -> {
+                        this.sleep();
+                        return "txt 1";
+                    },
+                    () -> {
+                        this.sleep();
+                        return "txt 2";
+                    },
+                    () -> {
+                        this.sleep();
+                        return "txt 3";
+                    }
+                ),
+                new HasValues<>("txt 1", "txt 2", "txt 3")
+            ).affirm()
+        );
+    }
+
+    /**
+     * Execution takes longer than timeout when
+     *  {@link ExecutorService} was initiated by {@link Timeout} itself.
+     */
+    @Test
+    public void failsDueToTimeoutWithInlineExecutorService() {
+        this.repeat(
+            arg -> new Assertion<>(
+                // @checkstyle LineLength (1 line)
+                "fails due to timeout when using the inline executor service",
+                () -> new Timeout<String>(
+                    3,
+                    1L,
+                    TimeUnit.MILLISECONDS,
+                    () -> {
+                        this.sleep();
+                        return "txt 1";
+                    },
+                    () -> {
+                        this.sleep();
+                        return "txt 2";
+                    },
+                    () -> {
+                        this.sleep();
+                        return "txt 3";
+                    }
+                ).iterator().next(),
+                new Throws<>(
+                    new IsEqual<>("java.util.concurrent.CancellationException"),
+                    CompletionException.class
+                )
+            ).affirm()
+        );
+    }
+
+    /**
+     * Repeat the test several times.
+     * @param test The test to execute.
+     */
+    private void repeat(final org.cactoos.Proc<?> test) {
+        new UncheckedFunc<>(
+            new Repeated<>(
+                arg -> {
+                    test.exec(null);
```

Comment:
@iakunin What's that `null`  references?

Reply:
@paulodamaso got rid of this `null`